models/posts.go

```diff
@@ -234,6 +243,15 @@ type PostArgs struct {
 	Author        map[string][]int64 `form:"author"`
 	Type          map[string][]int   `form:"type"`
 	Filter        Filter
+
+	// For filter API
+	FilterID          int64
+	FilterTitle       []string
+	FilterContent     []string
+	FilterAuthorName  []string
+	FilterTagName     []string
+	FilterPublishedAt map[string]time.Time
+	FilterUpdatedAt   map[string]time.Time
```

Comment:
How do you think about adding a `FilterNickname` or something like that to integrate the functionality of `GetIDsByNickname` with Filter API?

Anyway if we are going to integrate this function, it might need the collaboration from frontend.

Reply:
I see that was not hard to integrate with filter API since they both had similar behavior.
Maybe I can do it in this PR.

Reply:
After some inspect, `GetIDsByNickname` API takes extra parameters such as `role` and need to filter `active=1`.
So I guess we need to strike out another task to finish this job. lol

Reply:
It's just a thought. I know they have similar function, but may need a little bit different parameters. Let's put it in mind when we need to refactor it.